File: fisherYates.js
function fisherYates(a,currIndex)
{
  
  let randomIndex,temp;
  //prforming swaping in array
  for(let i=(currIndex-1);i>0;i--)
  {
      randomIndex=Math.floor(Math.random() * i);
      temp=a[currIndex];
      a[currIndex]=a[randomIndex];
      a[randomIndex]=temp;
    }
  return a;
}

//test
var inputArray = [];
var size; //Maximum Array size
size=prompt("enter size ");
for(var i=0; i<size; i++) {
	
	//Taking Input from user
	inputArray[i] = prompt('Enter Element ' + (i+1));
}
let result=fisherYates(inputArray,size-1);
console.log(result);

File: fisheryates.go
package main
import (
	"fmt"
	"math/rand"
)
func fisheryates(array []int,n int) []int{
	var temp int
	var random int
	for i:=n-1;i>0;i--{
		random=rand.Intn(n-1)
		temp=array[i]
		array[i]=array[random]
		array[random]=temp
	}
	return array
}
func main(){
	var n int
	var array []int
	fmt.Printf("Enter the size of the array: ")
	fmt.Scanf("%d",&n)
	array=make([]int,n)
	fmt.Printf("Enter the element in the array:\n")
	for i:=0;i<n;i++{
		fmt.Scanf("%d",&array[i])
	}
	res:=fisheryates(array,n)
	fmt.Println(res)
}
